pkg/api/search.go

```diff
@@ -78,7 +80,7 @@ func (hs *HTTPServer) Search(c *contextmodel.ReqContext) response.Response {
 		Limit:         limit,
 		Page:          page,
 		IsStarred:     starred == "true",
-		IsDeleted:     deleted == "true",
+		IsDeleted:     deleted == "true" && c.SignedInUser.GetOrgRole() == org.RoleAdmin,
```

Comment:
I think if you attempt to use the deleted filter as a non-admin, the API should just return a 401 error.